File: python/Programmers_level/level1/문자열 내 마음대로 정렬하기.py
# 파이썬은 그냥 lambda로 정렬하면 됨

# def solution(strings, n):
#     result = []
#     dict = {}
#     # 인덱스 문자 기준으로 dict에 저장
#     for i in strings:
#         if i[n] in dict:
#             dict[i[n]].append(i)
#         else:
#             dict[i[n]] = [i]
#     key = list(dict.keys())
#     key.sort()
    
#     # 오름차순으로 정렬한 결과 result에 저장
#     for j in key:
#         dict[j].sort()
#         for m in dict[j]:
#             result.append(m)

#     answer = result
#     return answer

def solution(strings, n):
    strings.sort()
    strings.sort(key=lambda x:x[n])
    return strings

File: Javascript/Programmers_level/level1/문자열 내 마음대로 정렬하기.js
function solution(strings, n) {
  strings.sort((a, b) => {
    let first = a[n];
    let second = b[n];
    // n번째 문자가 같을 경우
    if (first === second) {
      // 참이면 1, 거짓이면 -1 반환
      return (a > b) - (a < b);
    } else {
      return (first > second) - (first < second);
    }
  });

  return strings;
}

/*
function(a,b) < 0 이면 a를 b보다 작은 인덱스로 정렬한다.
function(a,b) == 0 이면 a와 b의 순서를 바꾸지 않는다.
function(a,b) > 0 이면 b를 a보다 작은 인덱스로 정렬한다.
*/
